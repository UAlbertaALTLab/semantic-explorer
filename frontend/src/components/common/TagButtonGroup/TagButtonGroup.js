import React from 'react';
// material ui button groups
import Button from '@mui/material/Button';
import ButtonGroup from '@mui/material/ButtonGroup';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import ClickAwayListener from '@mui/material/ClickAwayListener';
import Grow from '@mui/material/Grow';
import Paper from '@mui/material/Paper';
import Popper from '@mui/material/Popper';
import MenuItem from '@mui/material/MenuItem';
import MenuList from '@mui/material/MenuList';

const options = ['Default', 'Verb', 'Particle', 'Noun'];

var pos = 'Default';
export {pos};

// This is the component for the tag button group
// Using Material UI
export default function SplitButton() {
    const [open, setOpen] = React.useState(false);
    const anchorRef = React.useRef(null);
    const [selectedIndex, setSelectedIndex] = React.useState(0)

    const handleMenuItemClick = (event, index) => {
        setSelectedIndex(index);
        setOpen(false);
        if (options[index] == 'Verb') {             
            pos = 'V'    
        }
        if (options[index] == 'Noun') {             
            pos = 'N'         
        }         
        if (options[index] == 'Particle') {             
            pos = 'Ipc'         
        }         
        if (options[index] == 'Default') {             
            pos = 'Default'         
        }  
    };

    const handleToggle = () => {
        setOpen((prevOpen) => !prevOpen);
    };

    const handleClose = (event) => {
        if (anchorRef.current && anchorRef.current.contains(event.target)) {
        return;
        }

        setOpen(false);
    };

    // This is the main return statement for the tag button group component
    return (
        <React.Fragment>
        <ButtonGroup variant="contained" ref={anchorRef} aria-label="split button">
            <Button>{options[selectedIndex]}</Button>
            <Button
            size="small"
            aria-controls={open ? 'split-button-menu' : undefined}
            aria-expanded={open ? 'true' : undefined}
            aria-label="select merge strategy"
            aria-haspopup="menu"
            onClick={handleToggle}
            >
            <ArrowDropDownIcon />
            </Button>
        </ButtonGroup>
        <Popper
            sx={{
            zIndex: 1,
            }}
            open={open}
            anchorEl={anchorRef.current}
            role={undefined}
            transition
            disablePortal
        >
            {({ TransitionProps, placement }) => (
            <Grow
                {...TransitionProps}
                style={{
                transformOrigin:
                    placement === 'bottom' ? 'center top' : 'center bottom',
                }}
            >
                <Paper>
                <ClickAwayListener onClickAway={handleClose}>
                    <MenuList id="split-button-menu" autoFocusItem>
                    {options.map((option, index) => (
                        <MenuItem
                        key={option}
                        selected={index === selectedIndex}
                        onClick={(event) => handleMenuItemClick(event, index)}
                        >
                        {option}
                        </MenuItem>
                    ))}
                    </MenuList>
                </ClickAwayListener>
                </Paper>
            </Grow>
            )}
        </Popper>
        </React.Fragment>
    );
    }

