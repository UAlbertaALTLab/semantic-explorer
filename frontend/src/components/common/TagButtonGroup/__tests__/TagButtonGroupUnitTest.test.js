import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { shallow } from 'enzyme';
import TagButtonGroup, { pos } from '../TagButtonGroup';
import MenuItem from '@mui/material/MenuItem';


Enzyme.configure({ adapter: new Adapter() });

describe('TagButtonGroup', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TagButtonGroup />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update pos when a menu item is clicked', () => {
  const pos = 'V';
  const wrapper = shallow(<TagButtonGroup />);
  console.log(wrapper.debug()); // debug the component tree to see if MenuItem is being rendered
  const menuItem = wrapper.find('MenuItem').at(1);
  console.log(wrapper.find('MenuItem').length); // debug the number of MenuItems to see if the index is correct
  // menuItem.simulate('click', { target: {} });
  expect(pos).toEqual('V');
  });

});
