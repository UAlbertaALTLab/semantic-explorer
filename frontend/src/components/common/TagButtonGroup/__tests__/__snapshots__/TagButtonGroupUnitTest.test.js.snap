// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagButtonGroup should render correctly 1`] = `ShallowWrapper {}`;
